ts_sdk/task/__util_datalake.py

```
@@ -434,3 +521,23 @@ def create_labels_file(self, target_file, labels, sse_kms_key_id = None):
         response = self.s3.put_object(Body=json.dumps(labels), **params)
         print({ 'level': 'debug', 'message': 'labels file created', 'response': response })
 
+    # extra_args can only have specific key names. In this case, it cannot contain the
+    # keys: [Bucket, Key, CopySource]. These keys are already included in the copy() function signature.
+    def extract_extra_args_from_params(self, params):
+        extra_args = {}
+        for k, v in params.items():
+            if k in ['Bucket', 'Key', 'CopySource']:
+                continue
+            extra_args[k] = v
+        return extra_args
+
+    # We can't use s3.get_s3_head(file) because the file may have the old version before the copy command was called
+    # So we need to make a direct call to head_object without the versionId to get the latest version of the file
+    def get_latest_obj_version(self, bucket, file_key):
+        try:
+            new_head_obj = self.s3.head_object(Bucket=bucket, Key=file_key)
+            return new_head_obj.get('VersionId', '')
+        except Exception as e:
+            print({'level': 'error', 'message': 'could not retreive the latest version of the file because {}'.format(e)})
+            return ''
```

maybe better to rethrow the error?